marklogic-data-hub/src/main/java/com/marklogic/hub/impl/FlowManagerImpl.java

```diff
@@ -261,10 +264,16 @@ protected void deleteRelatedStepArtifacts(Flow flow, Step removedStep) {
             return;
         }
 
-        if (removedStep.isMappingStep() && !mappingIsReferencedByAFlow(removedStep)) {
+        StepDefinition.StepDefinitionType stepTypeOfRemovedStep = StepDefinition.StepDefinitionType.getStepDefinitionType(
+                                                    removedStep.getStepDefinitionType().toString()
+                                                    );
+
+        if (removedStep.isMappingStep() && !mappingIsReferencedByAFlow(removedStep) && !isCustomMapping(removedStep)) {
             deleteMappingArtifacts(flow, removedStep);
+        } else if(!removedStep.isCustomStep() && !stepIsReferencedByAFlow(removedStep.getName(), stepTypeOfRemovedStep)) {
+            deleteStepDefinitionArtifacts(removedStep, stepTypeOfRemovedStep);
         } else if (removedStep.isCustomStep() && !stepIsReferencedByAFlow(removedStep.getName(), StepDefinition.StepDefinitionType.CUSTOM)) {
-            deleteStepDefinitionArtifacts(removedStep);
+            deleteStepDefinitionArtifacts(removedStep, stepTypeOfRemovedStep);
```

Comment:
I recommend adding a test to FlowManagerServiceTest to cover the new scenario captured by the first "else if" block.

Reply:
Thanks Rob, I have it in my stash. Will pop it in here sometime later today or next week.

Reply:
@rjrudin I have pushed some tests. Pls review.